### PR TITLE
Add a warning to the CLI when a version conflict is detected

### PR DESCRIPTION
**Pull request by kevingosse**
With this change, the process check detects when the tracer is loaded multiple times and shows a warning. It also shows an error message if one of the Datadog.Trace.dll is version 1.x.

### CODE REVIEW

File: tracer/src/Datadog.Trace.Tools.Runner/Checks/Resources.cs [hunk 1]
```diff
@@ -26,6 +27,7 @@ internal static class Resources
         public const string GetProcessError = "Could not fetch information about target process. Make sure to run the command from an elevated prompt, and check that the pid is correct.";
         public const string IisNoIssue = "No issue found with the IIS site.";
         public const string IisMixedRuntimes = "The application pool is configured to host both .NET Framework and .NET Core runtimes. When hosting .NET Core, it's recommended to set '.NET CLR Version' to 'No managed code' to prevent conflicts.";
+        public const string VersionConflict = "Tracer version 1.x can't be loaded simultaneously with other versions and will produce orphaned traces. Make sure to synchronize the Datadog.Trace nuget version with the installed MSI version.";
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace.Tools.Runner/Checks/Resources.cs:30]:
```suggestion
        public const string VersionConflict = "Tracer version 1.x can't be loaded simultaneously with other versions and will produce orphaned traces. Make sure to synchronize the Datadog.Trace NuGet version with the installed MSI version.";
```

**Review by andrewlock — APPROVED** [review]
Nice addition 👍

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2448 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/9855be86bcb2f6ac47d634bc9f2222884480ed66) will **not change** line coverage
:heavy_check_mark: Merging #2448 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/9855be86bcb2f6ac47d634bc9f2222884480ed66) will **not change** branch coverage
:heavy_check_mark: Merging #2448 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/9855be86bcb2f6ac47d634bc9f2222884480ed66) will will **decrease** complexity by `2`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/9855be86bcb2f6ac47d634bc9f2222884480ed66) | #2448       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `11482` / `15868` | `11479` / `15867` |          |
| Lines %   | `72%`      | `72%`      |  `0%` :heavy_check_mark:  |
| Branches  | `6158` / `9272` | `6152` / `9270` |          |
| Branches %| `66%`      | `66%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `10475`      | `10473`        |  `-2`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/$dd-trace-dotnet/_build/results?buildId=90654&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/90654/artifacts?artifactName=Code%20Coverage%20Report_90654&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2448](https://dev.azure.com/datadoghq/$dd-trace-dotnet/_build/results?buildId=90724&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/90724/artifacts?artifactName=Code%20Coverage%20Report_90724&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/9855be86bcb2f6ac47d634bc9f2222884480ed66) | #2448       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `72%`      | `72%`       |  `0%` :heavy_check_mark:  |
| Branches %| `66%`      | `66%`       |  `0%` :heavy_check_mark:  |
| Complexity| `10475`      | `10473`       |  `-2` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Agent.StreamFactories.TcpStreamFactory](https://github.com/DataDog/dd-trace-dotnet/tree/ae6f0d2809093bdf9c03c3ca684915d5d54b8d56/tracer/src/Datadog.Trace/Agent/StreamFactories/TcpStreamFactory.cs) | `-90%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Telemetry.JsonHttpClientTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/ae6f0d2809093bdf9c03c3ca684915d5d54b8d56/tracer/src/Datadog.Trace/Telemetry/Transports/JsonHttpClientTelemetryTransport.cs) | `-26%` :no_entry: | `-38%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Agent.TracesTransportStrategy](https://github.com/DataDog/dd-trace-dotnet/tree/ae6f0d2809093bdf9c03c3ca684915d5d54b8d56/tracer/src/Datadog.Trace/Agent/TracesTransportStrategy.cs) | `-15%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.MongoDb.MongoDbIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/ae6f0d2809093bdf9c03c3ca684915d5d54b8d56/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbIntegration.cs) | `-7%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/$dd-trace-dotnet/_build/results?buildId=90654&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/90654/artifacts?artifactName=Code%20Coverage%20Report_90654&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2448](https://dev.azure.com/datadoghq/$dd-trace-dotnet/_build/results?buildId=90724&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/90724/artifacts?artifactName=Code%20Coverage%20Report_90724&api-version=6.0&%24format=zip)

**Review by zacharycmontoya — APPROVED** [review]
Nice 👍🏼

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2448 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f93b5bade586fbe1d2129f4c869f877f0125eb96) will **not change** line coverage
:heavy_check_mark: Merging #2448 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f93b5bade586fbe1d2129f4c869f877f0125eb96) will **not change** branch coverage
:heavy_check_mark: Merging #2448 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f93b5bade586fbe1d2129f4c869f877f0125eb96) will **not change** complexity


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/f93b5bade586fbe1d2129f4c869f877f0125eb96) | #2448       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `11540` / `15880` | `11493` / `15880` |          |
| Lines %   | `73%`      | `72%`      |  `0%` :heavy_check_mark:  |
| Branches  | `6198` / `9276` | `6167` / `9276` |          |
| Branches %| `67%`      | `66%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `10480`      | `10480`        |  `0`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/$dd-trace-dotnet/_build/results?buildId=90791&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/90791/artifacts?artifactName=Code%20Coverage%20Report_90791&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2448](https://dev.azure.com/datadoghq/$dd-trace-dotnet/_build/results?buildId=90891&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/90891/artifacts?artifactName=Code%20Coverage%20Report_90891&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/f93b5bade586fbe1d2129f4c869f877f0125eb96) | #2448       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `73%`      | `72%`       |  `0%` :heavy_check_mark:  |
| Branches %| `67%`      | `66%`       |  `0%` :heavy_check_mark:  |
| Complexity| `10480`      | `10480`       |  `0` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Telemetry.JsonWebRequestTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/8aaabf6dc0583e5ad99b955cdb9ca2890222d260/tracer/src/Datadog.Trace/Telemetry/Transports/JsonWebRequestTelemetryTransport.cs) | `-28%` :no_entry: | `-57%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.MongoDb.MongoDbIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/8aaabf6dc0583e5ad99b955cdb9ca2890222d260/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbIntegration.cs) | `-7%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/$dd-trace-dotnet/_build/results?buildId=90791&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/90791/artifacts?artifactName=Code%20Coverage%20Report_90791&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2448](https://dev.azure.com/datadoghq/$dd-trace-dotnet/_build/results?buildId=90891&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/90891/artifacts?artifactName=Code%20Coverage%20Report_90891&api-version=6.0&%24format=zip)

**Pull request merged**